extensions/standard-processors/processors/AttributesToJSON.cpp

```diff
@@ -0,0 +1,162 @@
+  core::PropertyBuilder::createProperty("Attributes List")
+    ->withDescription("Comma separated list of attributes to be included in the resulting JSON. "
+                      "If this value is left empty then all existing Attributes will be included. This list of attributes is case sensitive. "
+                      "If an attribute specified in the list is not found it will be be emitted to the resulting JSON with an empty string or NULL value.")
+    ->build());
+
+const core::Property AttributesToJSON::AttributesRegularExpression(
+  core::PropertyBuilder::createProperty("Attributes Regular Expression")
+    ->withDescription("Regular expression that will be evaluated against the flow file attributes to select the matching attributes. "
+                      "This property can be used in combination with the attributes list property.")
+    ->build());
+
+const core::Property AttributesToJSON::Destination(
+  core::PropertyBuilder::createProperty("Destination")
+    ->withDescription("Control if JSON value is written as a new flowfile attribute 'JSONAttributes' or written in the flowfile content. "
+                      "Writing to flowfile content will overwrite any existing flowfile content.")
+    ->isRequired(true)
+    ->withDefaultValue<std::string>("flowfile-attribute")
+    ->withAllowableValues<std::string>(DESTINATIONS)
+    ->build());
+
+const core::Property AttributesToJSON::IncludeCoreAttributes(
+  core::PropertyBuilder::createProperty("Include Core Attributes")
+    ->withDescription("Determines if the FlowFile core attributes which are contained in every FlowFile should be included in the final JSON value generated.")
+    ->isRequired(true)
+    ->withDefaultValue<bool>(true)
+    ->build());
+
+const core::Property AttributesToJSON::NullValue(
+  core::PropertyBuilder::createProperty("Null Value")
+    ->withDescription("If true a non existing or empty attribute will be NULL in the resulting JSON. If false an empty string will be placed in the JSON.")
+    ->isRequired(true)
+    ->withDefaultValue<bool>(false)
+    ->build());
+
+core::Relationship AttributesToJSON::Success("success", "All FlowFiles received are routed to success");
+
+void AttributesToJSON::initialize() {
+  setSupportedProperties({
+    AttributesList,
+    AttributesRegularExpression,
+    Destination,
+    IncludeCoreAttributes,
+    NullValue
+  });
+  setSupportedRelationships({Success});
+}
+
+void AttributesToJSON::onSchedule(core::ProcessContext* context, core::ProcessSessionFactory* /*sessionFactory*/) {
+  std::string attributes;
+  context->getProperty(AttributesList.getName(), attributes);
+  attribute_list_ = utils::StringUtils::split(attributes, ",");
+  context->getProperty(AttributesRegularExpression.getName(), attributes_regular_expression_str_);
+  if (!attributes_regular_expression_str_.empty()) {
+    attributes_regular_expression_ = utils::Regex(attributes_regular_expression_str_);
```

Comment:
If think we can already rely on std::regex, even if the c++20 PR is not yet merged. @szaszm can you confirm?

Moreover std::regex can throw in case of invalid regex. Which is fair as a throw in onschedule is going to cause a retry later, but I guess the error message is not going to be too talkative.

Reply:
Yes, std::regex is part of the language since C++11, and the reason we couldn't use it before was supporting GCC 4.8, which has only partial support for C++11. Now we have dropped support for anything older than GCC 8, so we are good with regex.

Reply:
Yes, when I opened this PR the GCC 4.8 support was still there, but I'll update this now.

Reply:
Updated in 78d69d19c56f1eaba7718b0253acdeba672f46f4